api/src/main/java/io/opentelemetry/api/baggage/ImmutableBaggage.java

```diff
@@ -135,6 +136,9 @@ public String toString() {
 
     @Override
     public Baggage.Builder put(String key, String value, EntryMetadata entryMetadata) {
+      if (!isKeyValid(key) || !isValueValid(value)) {
```

Comment:
Maybe we should do a fine-level logging here (can be discussed in/as a follow-up).

Reply:
why not throwing exception?

Reply:
We currently don't log for parameter checks in most places I believe. Maybe we should, but I think we should do it consistently separate from removing the exceptions.

@bogdandrutu To throw exceptions, I believe we'd need a spec change https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/error-handling.md#basic-error-handling-principles